### PR TITLE
Make call SchedulerServer::new once in ballista-scheduler process

### LINKED ISSUE #1531: Ballista scheduler dead loop in loadtest
**Describe the bug**
1. start scheduler `RUST_LOG=INFO cargo run   --bin ballista-scheduler` 
2. start one executor  'RUST_LOG=INFO cargo run --release  --bin ballista-executor '
3. run load test in #1521  with `RUST_LOG=INFO cargo run --bin tpch -- loadtest  ballista-load --query-list 1,3,5,6,7,10,12,13  --data-path ***/tpch-1g-oneFile/ --format parquet --host localhost --port 50050 --sql-path **` -> this run 100 sqls with 5 clients concurrency.

the scheduler will hang forever after  run cmd at most twice times.

**Additional context**
i use process sample to take the stack trace 

https://docs.google.com/document/d/13rTSlbE0NwCeZvw33dfJrhuVuPoc1RTWQSP42vlzoK8/edit?usp=sharing

it seems all tokio-runtime-worker are at `_pthread_cond_wait`

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?

Closes #1531 .

avoid  `tokio::spawn(async move { state_clone.synchronize_job_status_loop(). ` multi-times

### CODE REVIEW

**Comment by Ted-Jiang** [comment]
@houqp @alamb  Plz have a review.

**Review by houqp — APPROVED** [review]

File: ballista/rust/scheduler/src/main.rs [hunk 1]
```diff
@@ -62,14 +63,18 @@ async fn start_server(
         "Ballista v{} Scheduler listening on {:?}",
         BALLISTA_VERSION, addr
     );
+    //there should be only one SchedulerServer in process
```
Comment (alamb) [inline ballista/rust/scheduler/src/main.rs:66]: I don't understand how this code ensures just one `SchedulerServer` exists per process.

It seems to me what this code does is create a new `SchedulerServer` on a port that the operating system picks.

Comment (alamb) [reply ballista/rust/scheduler/src/main.rs:66]: Perhaps using something like `lazy_static` (https://docs.rs/lazy_static/1.4.0/lazy_static/, already a dependency) might be worth looking into

Comment (Ted-Jiang) [reply ballista/rust/scheduler/src/main.rs:66]: Thanks,  sorry for my wrong annotation. it should be `only call SchedulerServer::new() once in process ` ,
is this right ?

Comment (Ted-Jiang) [reply ballista/rust/scheduler/src/main.rs]: its ok to `clone()` SchedulerServer in multi-thread without call `tokio::spawn(async move { state_clone.synchronize_job_status_loop().await });`

Comment (houqp) [reply ballista/rust/scheduler/src/main.rs]: SchedulerServer::new creates the scheduler loop that runs in the background, that's the thing that's causing the issue.

Comment (Ted-Jiang) [reply ballista/rust/scheduler/src/main.rs]: @alamb Could you review it again, Thanks!

**Comment by alamb** [comment]
If it fixes your issues @Ted-Jiang  then it sounds good to me 👍

File: ballista/rust/scheduler/src/main.rs (hunk 1 above)
Comment (alamb) [reply ballista/rust/scheduler/src/main.rs]: Sorry -- I don't really know what this code is doing, so I just had an uninformed question; Sorry if this delayed merging unnecessarily.

**Pull request merged**